lib/config.h

```diff
@@ -46,6 +46,15 @@
 #  define NORETURN
 #endif
 
+// fallthrough
```

Comment:
Shouldn't https://en.cppreference.com/w/cpp/language/attributes/fallthrough be preferred if available?

Reply:
That is a C++-17 feature.

Reply:
Yes.
Anyway we may check if the current compiler support C++17 oder beyond...

Reply:
We not compile Cppcheck as C++17 at all so there's not really a point in adding that. Not sure if we even want to support that. @danmar What do you think?

Reply:
I would probably prefer to use `[[fallthrough]]` but as this is a define we can easily select the best fit and I am not against your code. I think it's ok to use this c++17 in development builds as long as cppcheck releases can be built with an old compiler. It's not crucial that developers use c++11 as long as we have CI that do use it and test it.

Reply:
feel free to change it later. I feel I will merge this as it is..